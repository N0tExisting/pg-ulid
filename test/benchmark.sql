BEGIN;

CREATE EXTENSION ulid;

-- Average time to generate 1.000.000 ULIDs.
WITH RECURSIVE avg_time_calc AS (
  SELECT interval '0' AS total_time, 0 AS iteration_count
  UNION ALL
  SELECT total_time + (clock_timestamp() - statement_timestamp()), iteration_count + 1
  FROM avg_time_calc
  WHERE iteration_count < 1000000
)
SELECT (SELECT total_time FROM avg_time_calc WHERE iteration_count = 1000000) / 1000000 AS avg_time;

-- Time to generate a single ULID.
WITH time_calc AS (
  SELECT clock_timestamp() - statement_timestamp() AS time_taken
)
SELECT time_taken FROM time_calc;

ROLLBACK;
