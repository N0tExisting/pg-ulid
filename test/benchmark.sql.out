BEGIN;
BEGIN
CREATE EXTENSION ulid;
CREATE EXTENSION
WITH RECURSIVE avg_time_calc AS (
  SELECT interval '0' AS total_time, 0 AS iteration_count
  UNION ALL
  SELECT total_time + (clock_timestamp() - statement_timestamp()), iteration_count + 1
  FROM avg_time_calc
  WHERE iteration_count < 1000000
)
SELECT (SELECT total_time FROM avg_time_calc WHERE iteration_count = 1000000) / 1000000 AS avg_time;
    avg_time     
-----------------
 00:00:00.189047
(1 row)

WITH time_calc AS (
  SELECT clock_timestamp() - statement_timestamp() AS time_taken
)
SELECT time_taken FROM time_calc;
   time_taken    
-----------------
 00:00:00.000086
(1 row)

ROLLBACK;
ROLLBACK
